src/include/conf.h

```diff
@@ -0,0 +1,1190 @@
+  WT_CONF_ID_os_cache_max,
+  WT_CONF_ID_overwrite,
+  WT_CONF_ID_prefix_compression,
+  WT_CONF_ID_prefix_compression_min,
+  WT_CONF_ID_prefix_search,
+  WT_CONF_ID_prepare_timestamp,
+  WT_CONF_ID_priority,
+  WT_CONF_ID_raw,
+  WT_CONF_ID_read_corrupt,
+  WT_CONF_ID_read_once,
+  WT_CONF_ID_read_timestamp,
+  WT_CONF_ID_readonly,
+  WT_CONF_ID_remove_files,
+  WT_CONF_ID_remove_shared,
+  WT_CONF_ID_salvage,
+  WT_CONF_ID_session_max,
+  WT_CONF_ID_session_scratch_max,
+  WT_CONF_ID_session_table_cache,
+  WT_CONF_ID_sessions,
+  WT_CONF_ID_skip_sort_check,
+  WT_CONF_ID_source,
+  WT_CONF_ID_split_deepen_min_child,
+  WT_CONF_ID_split_deepen_per_child,
+  WT_CONF_ID_split_pct,
+  WT_CONF_ID_stable_timestamp,
+  WT_CONF_ID_statistics,
+  WT_CONF_ID_strategy,
+  WT_CONF_ID_strict,
+  WT_CONF_ID_sync,
+  WT_CONF_ID_target,
+  WT_CONF_ID_terminate,
+  WT_CONF_ID_tiered_object,
+  WT_CONF_ID_tiers,
+  WT_CONF_ID_timeout,
+  WT_CONF_ID_timing_stress_for_test,
+  WT_CONF_ID_txn,
+  WT_CONF_ID_type,
+  WT_CONF_ID_use_environment,
+  WT_CONF_ID_use_environment_priv,
+  WT_CONF_ID_use_timestamp,
+  WT_CONF_ID_value_format,
+  WT_CONF_ID_verbose,
+  WT_CONF_ID_verify_metadata,
+  WT_CONF_ID_version,
+  WT_CONF_ID_write_through,
+  WT_CONF_ID_write_timestamp_usage,
+};
+/*
+ * Configuration key structure: END
+ */
+
+#define WT_CONF_BIND_VALUES_LEN 5
+
+/*
+ * WT_CONF_BINDINGS --
+ *	A set of values bound.
+ */
+struct __wt_conf_bindings {
+    u_int bind_values; /* position of top of values (next available) */
+
+    struct {
+        WT_CONF_BIND_DESC *desc;
+        WT_CONFIG_ITEM item;
+    } values[WT_CONF_BIND_VALUES_LEN];
+};
+
+/*
+ * WT_CONF_BIND_DESC --
+ *	A descriptor about a value to be bound.
+ */
+struct __wt_conf_bind_desc {
+    WT_CONFIG_ITEM_TYPE type; /* WT_CONFIG_ITEM.type */
+    u_int offset;             /* offset into WT_SESSION::conf_bindings.values table */
+};
+
+typedef enum {
+    CONF_COMPILED_SUBCONF = 0,
+    CONF_COMPILED_EXPLICIT = 1,
+    CONF_COMPILED_IMPLICIT = 2
+} WT_CONF_COMPILED_TYPE;
+
+/*
+ * WT_CONF --
+ *	A structure corresponding to a compiled configuration string.  This is set up to
+ * be a "position-independent" structure, so once set up, it can be copied without having
+ * to strdup any fields.
+ *
+ * We actually compile into a superstructure that has N WT_CONF struct in an array,
+ * followed by M WT_CONF_KEY structs in another array.  N is the maximum conf count,
+ * M is the maximum key count, these values are stored in this structure.  These two
+ * values are dependent on the particular API we are compiling, and whether the API has
+ * sub-configurations and how many keys the API uses.
+ *
+ * When we copy a conf (for example, by taking a precompiled configuration and making a
+ * copy so we can add modifications), we must copy the entire superstructure.
+ */
+struct __wt_conf {
+    /*
+     * Explicit compilation is allocated on the heap, implicit is allocated on the stack.
```

Comment:
Can I ask you for a short explainer (or a pointer to a comment that I might've missed) about the position-independence thing and and the way it interacts with explicit/implicit items and heap/stack allocation? I'm finding it hard to piece together from the code.